mesa/visualization/templates/js/GridDraw.js

```diff
@@ -140,15 +149,23 @@ const GridVisualization = function (
   this.drawCircle = function (
     x,
     y,
+    xAlign,
+    yAlign,
     radius,
     colors,
     stroke_color,
     fill,
     text,
     text_color
   ) {
-    const cx = (x + 0.5) * cellWidth;
-    const cy = (y + 0.5) * cellHeight;
+    // Prevent circle from being drawn outside cell bounds.
```

Comment:
I haven't checked this, but you are probably right. Because of the naming, that makes radius / 2 to be very confusing lol. We can clarify the code in a separate PR.